lib/libutils/isoc/include/malloc.h

```diff
@@ -86,37 +86,29 @@ void malloc_get_stats(struct pta_stats_alloc *stats);
 void malloc_reset_stats(void);
 #endif /* CFG_WITH_STATS */
 
-
 #ifdef CFG_NS_VIRTUALIZATION
-
-void nex_free(void *ptr);
-
 #ifdef ENABLE_MDBG
 
-void *nex_mdbg_malloc(const char *fname, int lineno, size_t size);
-void *nex_mdbg_calloc(const char *fname, int lineno, size_t nmemb, size_t size);
-void *nex_mdbg_realloc(const char *fname, int lineno, void *ptr, size_t size);
-void *nex_mdbg_memalign(const char *fname, int lineno, size_t alignment,
-			size_t size);
-
 void nex_mdbg_check(int bufdump);
 
-#define nex_malloc(size)	nex_mdbg_malloc(__FILE__, __LINE__, (size))
-#define nex_calloc(nmemb, size) \
-		nex_mdbg_calloc(__FILE__, __LINE__, (nmemb), (size))
-#define nex_realloc(ptr, size) \
-		nex_mdbg_realloc(__FILE__, __LINE__, (ptr), (size))
-#define nex_memalign(alignment, size) \
-		nex_mdbg_memalign(__FILE__, __LINE__, (alignment), (size))
-
+#define nex_malloc(size)	__mdbg_alloc(MAF_NEX, NULL, 1, 1, \
+					     (size), __FILE__, __LINE__)
+#define nex_calloc(nmemb, size)	__mdbg_alloc(MAF_NEX | MAF_ZERO_INIT, NULL, 1, \
+					     (nmemb), (size), __FILE__, \
+					     __LINE__)
+#define nex_realloc(ptr, size)	__mdbg_alloc(MAF_NEX, (ptr), 1, 1, \
+					     (size), __FILE__, __LINE__)
+#define nex_memalign(align, size) __mdbg_alloc(MAF_NEX, NULL, (align), 1, \
+					       (size), __FILE__, __LINE__)
 #else /* ENABLE_MDBG */
 
+#define nex_mdbg_check(x)        do { } while (0)
+
 void *nex_malloc(size_t size);
 void *nex_calloc(size_t nmemb, size_t size);
 void *nex_realloc(void *ptr, size_t size);
 void *nex_memalign(size_t alignment, size_t size);
-
-#define nex_mdbg_check(x)        do { } while (0)
+void nex_free(void *ptr);
```

Comment:
`nex_free()` needs to be declared whatever  `ENABLE_MDBG` value.

Reply:
Yes, you're right. I'll fix.